### PR TITLE
feat: add support for ipns and recursive to ipfs resolve

### PR DESCRIPTION
**Pull request by hugomrdias**
This PR adds IPNS support to resolve, makes the recursive option true by default and reworks the tests.

Jsdocs were added to the resolve method.

Two interface-core config profile tests needed to be skipped because js-ipfs doesn't support them yet

ref:
https://github.com/ipfs/js-ipfs/issues/2000

### CODE REVIEW

File: src/core/components/resolve.js [hunk 1]
```diff
@@ -1,64 +1,82 @@
...
-    }
+/**
+ * @typedef { import("../index") } IPFS
+ */
 
-    opts = opts || {}
+/**
+ * @typedef {Object} ResolveOptions
+ * @prop {string} cidBase - Multibase codec name the CID in the resolved path will be encoded with
+ * @prop {boolean} [recursive=true] - Resolve until the result is an IPFS name
```
Comment (dirkmc) [inline src/core/components/resolve.js]: Should this be `Resolve until the result is an IPFS key`?

File: src/core/components/resolve.js [hunk 2]
```diff
@@ -1,64 +1,82 @@
...
+
+/**
+ * IPFS Resolve factory
+ *
+ * @param {IPFS} ipfs
+ * @returns {ResolveWrapper}
+ */
+module.exports = (ipfs) => {
+  /**
+   * IPFS Resolve - Resolve the value of names to IPFS
```
Comment (dirkmc) [inline src/core/components/resolve.js]:
```suggestion
   * IPFS Resolve - Resolve names to IPFS keys
```

File: src/core/components/resolve.js (hunk 1 above)
Comment (dirkmc) [inline src/core/components/resolve.js]: Is the recursive parameter implemented? It seems from the code like it's assumed to always be true

File: src/http/api/resources/resolve.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ module.exports = {
   async handler (request, h) {
     const { ipfs } = request.server.app
     const name = request.query.arg
-    const recursive = request.query.r || request.query.recursive || false
+    const recursive = request.query.r || request.query.recursive || true
```
Comment (dirkmc) [inline src/http/api/resources/resolve.js]: If `request.query.recursive` is explicitly set to `false`, will this code result in `recursive` being set to `true`?

File: src/core/components/resolve.js (hunk 1 above)
Comment (hugomrdias) [reply src/core/components/resolve.js]: yeah probably or IPFS path, naming is broken in our code and docs :) 
right now this is just a copy paste from the docs for consistency

Comment (hugomrdias) [reply src/core/components/resolve.js]: right now this is just to be passed to ipns

File: src/http/api/resources/resolve.js (hunk 3 above)
Comment (hugomrdias) [reply src/http/api/resources/resolve.js]: hummm good catch

File: src/core/components/resolve.js (hunk 2 above)
Comment (hugomrdias) [reply src/core/components/resolve.js]: right now this is just a copy paste from the docs for consistency

File: src/core/components/resolve.js (hunk 1 above)
Comment (dirkmc) [reply src/core/components/resolve.js]: You're right IPFS path is more accurate 👍

File: package.json [hunk 4]
```diff
@@ -74,7 +74,6 @@
     "bs58": "^4.0.1",
     "buffer-peek-stream": "^1.0.1",
     "byteman": "^1.3.5",
-    "callbackify": "^1.1.0",
```
Comment (alanshaw) [inline package.json]: `callbackify` is used here https://github.com/ipfs/js-ipfs/blob/813048ffb32a11cfefc51c2ec8634faaff2a924e/src/core/components/files-mfs.js so will need to be migrated to `promise-nodeify`. Would be rad if you also migrated `libp2p-kad-dht` also 😉🙏:

```
ipfs@0.37.0-rc.0 /Users/alan/Code/pl/ipfs/js-ipfs
├── callbackify@1.1.0 
└─┬ libp2p-kad-dht@0.15.3
  └── callbackify@1.1.0  deduped
```

**Comment by alanshaw** [comment]
BREAKING CHANGE: `recursive` is now `true` by default in `ipfs resolve`

**Comment by alanshaw** [comment]
@hugomrdias 🙏 please can we have this one fixed up?

**Comment by alanshaw** [comment]
blocked by https://github.com/ipfs/js-ipfs/issues/2416

**Comment by alanshaw** [comment]
Ok, DNS on ipfs.io has been fixed - @hugomrdias can you fix up the other test failures please? 🙏

**Comment by hugomrdias** [comment]
ready

**Pull request merged**